src/StreamingBodyParser/RawBodyParser.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace React\Http\StreamingBodyParser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+use React\Promise\ExtendedPromiseInterface;
+
+class RawBodyParser implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    /**
+     * @var ExtendedPromiseInterface
+     */
+    private $promise;
+
+    /**
+     * @param Request $request
+     * @return ParserInterface
+     */
+    public static function create(Request $request)
+    {
+        return new static($request);
+    }
+
+    /**
+     * @param Request $request
+     */
+    private function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->promise = ContentLengthBufferedSink::createPromise(
+            $request,
+            $headers['content-length']
+        )->then([$this, 'finish']);
```

Comment:
Ideally, this should be `done()`. Since you're supporting react/promise < ^2.2, i'm not sure if you there should be a `method_exists` check here.